src/store/reducers.js

```diff
@@ -0,0 +1,33 @@
+import { createVehicle } from "./helpers";
+
+export const addVehicleReducer = (state) => {
+  return {
+    ...state,
+    1: { ...state[1], vehicle: createVehicle(state.nextVehicle, 1) },
+    nextVehicle: state["nextVehicle"] + 1
+  };
+};
+
+export const updateVehicleStatusReducer = (state, station, newStatus) => {
+  return {
+    ...state,
+    [station]: {
+      ...state[station],
+      vehicle: { ...state[station].vehicle, state: newStatus }
+    }
+  };
+};
+
+export const moveVehicleReducer = (state, current_station) => {
+  const next_station = parseInt(current_station) + 1;
+  const vehicle = createVehicle(
+    state[current_station].vehicle.id,
+    next_station
+  );
+
+  return {
+    ...state,
+    [current_station]: { ...state[current_station], vehicle: null },
+    [next_station]: { ...state[next_station], vehicle }
+  };
```

Comment:
💡 The business logic to make sure that the next station can accommodate this vehicle is handled in the component itself. Here we assume that a call that has reached this reducer is valid, and hence we move our vehicle to the next stations as well as  empty our current station.